src/apatheia/jobs.nim

```diff
@@ -34,6 +34,22 @@ type
     id*: JobId
     queue*: SignalQueue[(JobId, T)]
 
+  OpenArrayHolder*[T] = object
+    data*: ptr UncheckedArray[T]
+    size*: int
+
+type
+  MemoryHolder* = ref object of RootObj
```

Comment:
@gmega @tbekas  something like these, just polymorphic containers. All they need to be is a ref object. Any seq's or string's can be moved into them, and held until the task is done.